x-pack/plugins/cases/server/client/configure/client.ts

```diff
@@ -87,50 +107,97 @@ export const createConfigurationSubClient = (
+      excess(GetConfigureFindRequestRt).decode(params),
+      fold(throwErrors(Boom.badRequest), identity)
+    );
+
+    const {
+      filter: authorizationFilter,
+      ensureSavedObjectsAreAuthorized,
+      logSuccessfulAuthorization,
+    } = await getAuthorizationFilter({
+      authorization,
+      operation: Operations.findConfigurations,
+      auditLogger,
+    });
+
+    const filter = combineAuthorizedAndOwnerFilter(
+      queryParams.owner,
+      authorizationFilter,
+      Operations.findConfigurations.savedObjectType
+    );
+
     let error: string | null = null;
+    const myCaseConfigure = await caseConfigureService.find({
+      soClient,
+      options: { filter },
+    });
+
+    ensureSavedObjectsAreAuthorized(
+      myCaseConfigure.saved_objects.map((configuration) => ({
+        id: configuration.id,
+        owner: configuration.attributes.owner,
+      }))
+    );
 
-    const myCaseConfigure = await caseConfigureService.find({ soClient });
+    logSuccessfulAuthorization();
 
-    const { connector, ...caseConfigureWithoutConnector } = myCaseConfigure.saved_objects[0]
-      ?.attributes ?? { connector: null };
-    let mappings: ConnectorMappingsAttributes[] = [];
-    if (connector != null) {
-      try {
-        mappings = await casesClientInternal.configuration.getMappings({
-          connectorId: connector.id,
-          connectorType: connector.type,
-        });
-      } catch (e) {
-        error = e.isBoom
-          ? e.output.payload.message
-          : `Error connecting to ${connector.name} instance`;
-      }
-    }
+    const configurations = await Promise.all(
+      myCaseConfigure.saved_objects.map(async (configuration) => {
+        const { connector, ...caseConfigureWithoutConnector } = configuration?.attributes ?? {
+          connector: null,
+        };
+
+        let mappings: SavedObjectsFindResponse<ConnectorMappings>['saved_objects'] = [];
 
-    return myCaseConfigure.saved_objects.length > 0
-      ? CaseConfigureResponseRt.encode({
+        if (connector != null) {
+          try {
+            mappings = await casesClientInternal.configuration.getMappings({
+              connectorId: connector.id,
+              connectorType: connector.type,
+            });
+          } catch (e) {
+            error = e.isBoom
+              ? e.output.payload.message
+              : `Failed to retrieve mapping for ${connector.name}`;
+          }
+        }
+
+        return {
           ...caseConfigureWithoutConnector,
           connector: transformESConnectorToCaseConnector(connector),
-          mappings,
-          version: myCaseConfigure.saved_objects[0].version ?? '',
+          mappings: mappings.length > 0 ? mappings[0].attributes.mappings : [],
+          version: configuration.version ?? '',
           error,
-        })
-      : {};
+          id: configuration.id,
+        };
+      })
+    );
+
+    return CaseConfigurationsResponseRt.encode(configurations);
```

Comment:
I thought based on @XavierM comments we wanted to only return a single value in the response to the UI?

Reply:
For the `get` we aggreed it will return an array because users with permissions to multiple solutions should get one configuration per solution. For the UI, it will be an array of one elemnt.